test/python/onnx_importer/command_line_test.py

```diff
@@ -0,0 +1,145 @@
+        value=numpy_helper.from_array(numpy.array([4], dtype=numpy.int64)))
+    cofshape = make_node(
+        'ConstantOfShape', ['c_shape'], ['c_out'], 'cofshape',
+        value=numpy_helper.from_array(numpy.array([1], dtype=numpy.int64)))
+
+    outval = make_tensor_value_info('c_out', TensorProto.INT64, [None])
+    graph = make_graph([const, cofshape], 'constgraph', [], [outval])
+
+    onnx_model = make_model(graph)
+    check_model(onnx_model)
+    return onnx_model
+
+
+def linear_model() -> onnx.ModelProto:
+    # initializers
+    k_dim = 32
+    value = numpy.arange(k_dim).reshape([k_dim, 1])
+    value = numpy.asarray(value, dtype=numpy.float32)
+    A = numpy_helper.from_array(value, name='A')
+
+    value = numpy.array([0.4], dtype=numpy.float32).reshape([1, 1])
+    C = numpy_helper.from_array(value, name='C')
+
+    # the part which does not change
+    X = make_tensor_value_info('X', TensorProto.FLOAT, [1, k_dim])
+    Y = make_tensor_value_info('Y', TensorProto.FLOAT, [None, None])
+    node1 = make_node('MatMul', ['X', 'A'], ['AX'])
+    node2 = make_node('Add', ['AX', 'C'], ['Y'])
+    graph = make_graph([node1, node2], 'lr', [X], [Y], [A, C])
+    onnx_model = make_model(graph)
+    check_model(onnx_model)
+    return onnx_model
+
+
+ALL_MODELS = [
+    const_model,
+    linear_model
+]
+
+
+class CommandLineTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.test_dir = OUTPUT_PATH / "command-line"
+        shutil.rmtree(cls.test_dir, ignore_errors=True)
+        cls.test_dir.mkdir(parents=True, exist_ok=True)
+
+    def get_run_path(self, model_name: str) -> Path:
+        run_path = CommandLineTest.test_dir / model_name
+        run_path.mkdir(exist_ok=True)
+        return run_path
+
+    def run_model_intern(self, onnx_model: onnx.ModelProto, model_name: str):
+        run_path = self.get_run_path(model_name)
+        model_file = run_path / f"{model_name}-i.onnx"
+        mlir_file = run_path / f"{model_name}-i.torch.mlir"
+        onnx.save(onnx_model, model_file)
+        p = subprocess.run([
+            sys.executable, "-m", "torch_mlir.tools.import_onnx", model_file,
+            "-o", mlir_file])
+        self.assertEqual(p.returncode, 0)
+
+    def run_model_extern(self, onnx_model: onnx.ModelProto, model_name: str):
+        run_path = self.get_run_path(model_name)
+        model_file = run_path / f"{model_name}-e.onnx"
+        mlir_file = run_path / f"{model_name}-e.torch.mlir"
+        data_dir_name = f"{model_name}-data"
+        model_data_dir = run_path / data_dir_name
+        model_data_dir.mkdir(exist_ok=True)
+        convert_model_to_external_data(
+            onnx_model, all_tensors_to_one_file=True,
+            location=data_dir_name + "/data.bin",
+            size_threshold=48,
+            convert_attribute=True)
+        onnx.save(onnx_model, model_file)
+        temp_dir = run_path / "temp"
+        temp_dir.mkdir(exist_ok=True)
+        p = subprocess.run([
```

Comment:
Pro tip: For these cases, I often don't use a literal sub-process but just call into the main entrypoint from importing the model, giving it explicit command line arguments.

(can ignore / I might fix in a followup)

Reply:
Little more clarity since you asked offline:

```
from torch_mlir.tools.import_onnx import __main__

...


args = __main__.parse_arguments([1, 2, 3])
__main__.main(args)
```

Something like that. Often if I'm doing it, I'll have an entry point in the module just for testing instead of needing to poke in in two steps like that.